File: src/main/java/my/study/repository/design/pattern/proxy/UseDemo.java
package my.study.repository.design.pattern.proxy;

/**
 * @author djh on  2018/9/7 13:43
 * @E-Mail 1544579459@qq.com
 */
public class UseDemo {
    public static void main(String[] args) {
        RealImp realImp = new RealImp();
        ProxyMen proxyMen = new ProxyMen(realImp);
        // Use the proxy men to visit the realImp someMethod.
        proxyMen.someMethod();
    }
}

File: src/main/java/my/study/repository/design/pattern/proxy/kotlin/UseDemo.kt
package my.study.repository.design.pattern.proxy.kotlin

/**
 * @author djh on  2018/10/15 19:24
 * @E-Mail 1544579459@qq.com
 */
object UseDemo {
    @JvmStatic
    fun main(args: Array<String>) {
        val proxyMen = ProxyMen()
        proxyMen.someMethod()
    }
}